massgov/mayflower
# Core/usage and architecture doc updates

## Description
After our discussion with Bryan I felt like we needed more detail on how Mayflower can be used so I added some content.

## Steps to Test
1. Pull the branch
2. Run  rush start:core
3. Review the pages in the browser
4. Check links
